data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/transformer/DataPostProcessor.java

```diff
@@ -10,20 +10,34 @@
 
 import java.util.Arrays;
 import java.util.Comparator;
+import java.util.List;
+import java.util.TreeMap;
 import java.util.function.Function;
 import java.util.function.Predicate;
 
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
 @Slf4j
 @NoArgsConstructor
 public class DataPostProcessor {
     UtilHelper utilHelper = UtilHelper.getInstance();
 
     public <T extends PersonExtendedProps> void processPersonName(String name, T pf) {
         if (!ObjectUtils.isEmpty(name)) {
-            Arrays.stream(utilHelper.deserializePayload(name, Name[].class))
+            List<Name> nameList = Arrays.stream(utilHelper.deserializePayload(name, Name[].class))
                     .filter(pName -> !ObjectUtils.isEmpty(pName.getPersonUid()))
```

Comment:
Adding in a comment for integration of nm_use_cd in logic.

Reply:
Updated the logic for the `nm_use_cd`
It is applicable only for the Patient i.e., Legal and alias.
For provider only Legal names.